backend/routes/webhooks/webhooks.go

```diff
@@ -0,0 +1,81 @@
+package routes
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"strings"
+
+	"github.com/emicklei/go-restful/v3"
+	"github.com/google/go-github/github"
+)
+
+type WebHookInfo struct {
+	Id        string // Id for the github webhook request
+	Event     string // indicates which event took place (push, starred, pull request etc)
+	Signature string // signature of the github webhook request
+	Payload   []byte // consists of all the contents within the webhook
+}
+
+func ParseWebhookInfo(request *restful.Request, response *restful.Response) {
```

Comment:
We can add an `error` as the return value for this function, and convert all the current `log.Fatalln(...)` (and others) calls to `return fmt.Errorf(...)` calls.

Reply:
As per the research, go-restful route function won't allow us to have an error as a return value!

The struct for the restful.RouteFunction is defined as within the module:
`type RouteFunction func(*Request, *Response)`

It shows error on manipulating details, so that was the finding. If there is any other approach to this please let me know :D